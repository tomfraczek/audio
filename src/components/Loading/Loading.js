// @flow
import React from 'react';
import type {Node} from 'react';
import styles from './Loading.module.css';
import logo from '../../assets/spinner.png';

const Loading = ({text}: {text: string}): Node => (
  <div className={styles.loading}>
    <div className={styles.overlay}>
      <div className={styles.spinner}>
        <img src={logo} alt={'Logo spinning'} />
      </div>
      <div className={styles.text}>
        <h2>{text}</h2>
      </div>
    </div>
  </div>
);

export default Loading;
