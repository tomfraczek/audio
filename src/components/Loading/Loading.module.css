.loading {
  position: fixed;
  top: 0;
  left: 0;
  z-index: 1;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100%;
  width: 100%;
  z-index: 999;
}

.loading:before {
  content: '';
  height: 100%;
  width: 100%;
  position: absolute;
  top: 0;
  left: 0;
  background: #333;
  opacity: 0.4;
  z-index: 998;
}

.overlay {
  position: relative;
  z-index: 1000;
  text-align: center;
}

.text {
  padding: 25px;
}

.spinner img {
  animation: rotate 2s infinite linear;
  max-width: 200px;
  max-height: 200px;
}

@keyframes rotate {
  0%   { transform: rotate(0); }
  100% { transform: rotate(360deg); }
}
