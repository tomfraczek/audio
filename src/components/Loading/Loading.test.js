import React from 'react';
import Loading from './Loading';
import renderer from 'react-test-renderer';

test('Has loading thing', () => {
  const component = renderer.create(
    <Loading text="Works" />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
