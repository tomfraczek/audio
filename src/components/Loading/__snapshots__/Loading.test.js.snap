// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Has loading thing 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    <div
      className={undefined}
    >
      <img
        alt="Logo spinning"
        src="spinner.png"
      />
    </div>
    <div
      className={undefined}
    >
      <h2>
        Works
      </h2>
    </div>
  </div>
</div>
`;
